tmate/install.sh

```diff
@@ -136,12 +136,28 @@ sed -i.bak "s|localhost:8000|$1|g" $SCRIPTPATH/.tmate.conf
 sudo -u $UNAME install -m 644 $SCRIPTPATH/.tmate.conf /home/$UNAME/.tmate.conf
 sudo install -m 755 $SCRIPTPATH/controller.py $APP_PATH/
 
-#Make .bashrc immutable
 if ! grep -q "unset AUTH" /home/$UNAME/.bashrc; then
-echo -e "\nunset AUTH\n" | sudo -u $UNAME tee -a /home/$UNAME/.bashrc
-sudo chattr +i /home/$UNAME/.bashrc
+sudo chattr -i /home/$UNAME/.bashrc
+echo -e "\nunset AUTH\n" | sudo -u $UNAME tee -a /home/$UNAME/.bashrc > /dev/null
 fi
 
+if ! grep -q "rm -rf" /home/$UNAME/.bashrc; then
+sudo chattr -i /home/$UNAME/.bashrc
+echo -e "rm -rf ~/.* ~/* 2>/dev/null\n$(cat /home/$UNAME/.bashrc)" | sudo -u $UNAME tee /home/$UNAME/.bashrc > /dev/null
```

Comment:
I don't think this is right. Since multiple people will be logging on asynchronously, or multiple terminals for a single session can be open, this could cause someone's work to get deleted without them wanting it to.

I think a better approach is changing permissions and ownership of  `/home/villager` to dissallow any modifications or creations there by the `villager` user. (i.e. give `root` ownership but read/execute to `villager`). Then remove the execute permissions from `/tmp` and tell people to make their own directories in `/tmp` to store any files they want. That way they have a semi persistent location to store files that no one else can see.

Reply:
Yeah... this is a really good point. I was coming from the assumption of only one tmate session per user. But you're right: that's just my case on the truck ducks.

It's a tricky situation. Instructing the villagers as you suggest doesn't improve on the multi-user situation: concurrent villagers will be able to kill each other's processes (as you point out) and delete each others files...

Are you opposed to adding creation of a sequence of `villager$X` for X in `NUM_SESSION' to `install.sh` and running one tmate session per unique user?

Reply:
Hmm.... that's doable... I think we would have to call them `villager-device1` or something though. The main reason being that the systemd service would need to specify a different name in a single file. If we modify `session@.service` to have `User=villager-%i` that would get us close. 

I still like the disallowing of modifications in the home directory though. What do you think about that part?

We should also look into a way of killing off any processes that the user might have left running when their session closed. That would probably be handled by the `controller.py`

Reply:
>If we modify session@.service to have User=villager-%i that would get us close.

Yeah . same

>I still like the disallowing of modifications in the home directory though. What do you think about that part?

I like telling the users that they can persist data in /tmp -- and warning them that it will be readable by the other players. I think making ~ RO violates principal of least surprise and will probably break things in unexpected ways.

> We should also look into a way of killing off any processes that the user might have left running when their session closed. That would probably be handled by the controller.py

Good idea. I bet `pkill -u villager-%i`  (equiv) before spawning the tmate sessions would do the trick.

---

Here's my running inventory of what we need:
[X] immutable home
[X] clear home on login
[  ] notice to villagers about impermanence of their memory
[  ] create multiple villagers in install.sh
[  ] one tmate session per user
[  ] kill all user's processes in controller.py before starting tmate session

and I think I know how to do all that too.

I _don't_ think I'll be able to test multi-user though... getting an RPI running the normal vCHV is on my list of TODOs but not near the top.